app/helpers/StringHelper.php

```diff
@@ -392,3 +412,86 @@ function urilize($path, bool $noTime = false): string
 function implodeCsv($value) {
     return implode(', ', $value);
 }
+
+/**
+ * Returns true if the value is an IPv6 or IPv4 address (in that order)
+ */
+function isIpAddress($value) {
+    $ipV6Regex = '/^\[(?:(?:(?:[0-9A-Fa-f]{0,4}:){7}[0-9A-Fa-f]{0,4})|(?:(?:[0-9A-Fa-f]{0,4}:){6}:[0-9A-Fa-f]{0,4})|(?:(?:[0-9A-Fa-f]{0,4}:){5}:(?:[0-9A-Fa-f]{0,4}:)?[0-9A-Fa-f]{0,4})|(?:(?:[0-9A-Fa-f]{0,4}:){4}:(?:[0-9A-Fa-f]{0,4}:){0,2}[0-9A-Fa-f]{0,4})|(?:(?:[0-9A-Fa-f]{0,4}:){3}:(?:[0-9A-Fa-f]{0,4}:){0,3}[0-9A-Fa-f]{0,4})|(?:(?:[0-9A-Fa-f]{0,4}:){2}:(?:[0-9A-Fa-f]{0,4}:){0,4}[0-9A-Fa-f]{0,4})|(?:(?:[0-9A-Fa-f]{0,4}:){6}(?:(?:(?:25[0-5])|(?:2[0-4]\d)|(?:1\d{2})|(?:\d{1,2}))\.){3}(?:(?:25[0-5])|(?:2[0-4]\d)|(?:1\d{2})|(?:\d{1,2})))|(?:(?:[0-9A-Fa-f]{0,4}:){0,5}:(?:(?:(?:25[0-5])|(?:2[0-4]\d)|(?:1\d{2})|(?:\d{1,2}))\.){3}(?:(?:25[0-5])|(?:2[0-4]\d)|(?:1\d{2})|(?:\d{1,2})))|(?:::(?:[0-9A-Fa-f]{0,4}:){0,5}(?:(?:(?:25[0-5])|(?:2[0-4]\d)|(?:1\d{2})|(?:\d{1,2}))\.){3}(?:(?:25[0-5])|(?:2[0-4]\d)|(?:1\d{2})|(?:\d{1,2})))|(?:[0-9A-Fa-f]{0,4}::(?:[0-9A-Fa-f]{0,4}:){0,5}[0-9A-Fa-f]{0,4})|(?:::(?:[0-9A-Fa-f]{0,4}:){0,6}[0-9A-Fa-f]{0,4})|(?:(?:[0-9A-Fa-f]{0,4}:){1,7}:))\]$/';
+    $ipV4Regex = '/^(?:25[0-5]|2[0-4]\d|1\d\d|[1-9]\d|\d)(?:\.(?:25[0-5]|2[0-4]\d|1\d\d|[1-9]\d|\d)){3}$/';
+
+    return (preg_match($ipV6Regex, $value) || preg_match($ipV4Regex, $value));
+}
+
+/**
+ * Returns a part of a JID, stringprep'ed with the given profile,
+ * or false if it is invalid under that profile.
+ */
+function prepJidPart($value, $prep) {
+    try {
+        $ret = $prep->apply($value);
+    } catch (ProfileException $e) {
+        return false;
+    }
+
+    if ((strlen($ret) < 1) || (strlen($ret) > 1023)) {
+        return false;
+    }
+
+    return $ret;
+}
+
+/**
+ * Returns a JID, stringprep'ed with the given profile,
+ * or false if it is invalid under that profile.
+ */
+function prepJid($value) {
+    // Split JID into parts
+    $jid_parts = explodeJid($value);
+
+    // Validate local part against Nodeprep profile
+    if ($jid_parts['username'] !== '') {
+        $jid_parts['username'] = prepJidPart($jid_parts['username'], new Nodeprep());
+        if ($jid_parts['username'] === false) {
+            return false;
+        }
+    }
+
+    // Validate resource part against Resourceprep profile
+    if ($jid_parts['resource'] !== '') {
+        $jid_parts['resource'] = prepJidPart($jid_parts['resource'], new Resourceprep());
+        if ($jid_parts['resource'] === false) {
+            return false;
+        }
+    }
+
+    // Finally, validate domain part (IPv6, IPv4, FQDN, in that order)
+    if (!isIpAddress($jid_parts['server'])) {
```

Comment:
Please be consistent, could you change all the variables to camelCase instead of snake_case. 
I prefer to use camelCase in Movim :)